### PR TITLE
docs: add ADR and update changelog

### PR DESCRIPTION
**Pull request by MoisesGSalas**
## Description

Include the Architectural Decision Record for the changes made #153

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -12,6 +12,11 @@ Change Log
 Unreleased
 ----------
```
Comment (magajh) [inline CHANGELOG.rst:14]: Shouldn't the updated version and date be up here?

Comment (MoisesGSalas) [reply CHANGELOG.rst:14]: We add the date once we are ready to release. The next release is going to also include PR#155 so I added my changes to make it easier for whoever has to make the new release.

File: docs/decisions/0001-include-slug-in-uid.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+--------
+A rather simple solution was already present in the saml backend, which prepends the slug to de uid. Therefore we have
+decided to do the same for the other providers.
+
+The design
+===========
+We alter the ``get_user_id`` function from the generic OIDC backend to return a modified version of the uid that includes
+the slug in the format ``slug:uid``. The slug is obtained from the provider configuration and must be set under the
+``Other Settings`` field, as this is the only field the backend can access directly besides ID and Secret.
+All new associations will be correctly namespaced with the given slug. For existing associations the new
```
Comment (mariajgrimaldi) [inline docs/decisions/0001-include-slug-in-uid.rst:24]: why don't you add For existing associations... to the consequences section?

File: docs/decisions/0001-include-slug-in-uid.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Through the configuration values ``SOCIAL_AUTH_NAMESPACED_UIDS``, ``SOCIAL_AUTH_ALLOW_SLUGLESS_UID`` and
+``SOCIAL_AUTH_ALLOW_WRITE_SLUG_UID`` one can chose to enable the feature, forbid the use of the old format
+and update the previous associations to the new format at login time respectively.
+
+This changes will have to eventually be applied to other backends in the future.
+
+Rejected alternatives
+---------------------
+1.
```
Comment (mariajgrimaldi) [inline docs/decisions/0001-include-slug-in-uid.rst:36]: are you using the hyphens or the numbers

Comment (MoisesGSalas) [reply docs/decisions/0001-include-slug-in-uid.rst:36]: Both: 
image

File: docs/decisions/0001-include-slug-in-uid.rst (hunk 2 above)
Comment (MoisesGSalas) [reply docs/decisions/0001-include-slug-in-uid.rst:24]: The 'For existing...' part is supposed to explain how those existing associations are handled. So it seemed best have it there.

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**